src/compojure/api/common.clj

```diff
@@ -41,3 +41,11 @@
   (if (get v 1)
     (apply merge v)
     (get v 0)))
+
+(defn fast-map-merge
```

Comment:
Ah well, turns out TransientArrayMap isn't as optimized as PersistentArrayMap, so even though it does less work than PersistentArrayMap, it's slower. With a small patch to Clojure you could make them equally fast in this benchmark! :P